src/main/java/com/appland/appmap/output/v1/CodeObject.java

```diff
@@ -319,4 +342,12 @@ public CodeObject addChild(CodeObject child) {
 
     return this;
   }
+
+  public ArrayList<CodeObject> getChildren() {
+    // Once, among many runs, I saw a ConcurrentAccessException on
+    // this.children. I wasn't able to reproduce it. If it comes back,
+    // though, it may be necessary to return a copy of it. 20200824, ajp
+    // return new ArrayList<CodeObject>(this.children);
+    return this.children;
```

Comment:
Try returning Collections.synchronizedList

Reply:
Do you think I should address this, even though I haven't been able to reproduce it?

The ConcurrentAccessException was thrown from an iterator, not from the array list. This: https://stackoverflow.com/a/53614929 suggests that `Collections.synchronizedList` doesn't address that. Seems like just returning a shallow copy would though, no?

Reply:
Hmm. Perhaps this is the cause, and if it is I believe the call site can be synchronized to avoid making `CodeObject`/`CodeObjectTree` thread safe:
https://github.com/applandinc/appmap-java/blob/less-mem_20200817/src/main/java/com/appland/appmap/record/Recorder.java#L110-L113

In short, `writeEvent` used to be synchronized. It's no longer synchronized due to the fact that we can encounter a lock when calling `toString` into the client application. It's certainly worth looking into. I'll defer to you @apotterri if you'd like to look into it now or as a follow on item.